libsql-server/src/namespace/meta_store.rs

```diff
@@ -399,7 +400,58 @@ impl MetaStore {
         wal_manager: MetaStoreWalManager,
     ) -> Result<Self> {
         let (changes_tx, mut changes_rx) = mpsc::channel(256);
-        let inner = Arc::new(MetaStoreInner::new(base_path, conn, wal_manager, config).await?);
+
+        let destroy_on_error = config.destroy_on_error;
+
+        let inner = match MetaStoreInner::new(base_path, conn, wal_manager, config.clone()).await {
+            Ok(inner) => inner,
+            Err(e) => {
+                if destroy_on_error {
+                    let db_path = base_path.join("metastore");
+
+                    tracing::info!(
```

Comment:
We should probably have a warning-level log telling us this operation will happen or is done, so we can detect it easily.